DirectXHost.cs

```diff
@@ -8,230 +8,261 @@
+
+                _graphics = new GraphicsD3D11();
+                _graphics.Initialise(_dxForm, true);
+                _dxForm.UserResized += (sender, args) =>
+                {
+                    var renderForm = sender as RenderForm;
+                    ClientSize = new Size(renderForm.ClientSize.Width, renderForm.ClientSize.Height);
+                    UserResized = true;
+                };
+                _overlayForm.Show();
+                RenderLoop.Run(_dxForm, RenderCallback, true);
+            }
+            catch (ArgumentException ex)
+            {
+                MessageBox.Show(string.Format("Exception running DirectX host window.\r\n{0}", ex.Message), "DirectX Host", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(string.Format("Exception running DirectX host window.\r\n{0}", ex.Message), "DirectX Host", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+        }
+
+        // Set the bitmap object to the size of the screen
+        private Bitmap _bmpScreenshot;
+        Bitmap bmpScreenshot
+        {
+            get => _bmpScreenshot;
+            set
+            {
+                _bmpScreenshot = value;
+                _overlayTarget.Image = value;
+                updateImageBounds();
+            }
+        }
+
+        Graphics gfxScreenshot;
+        private void RenderCallback()
+        {
+            if (_dxForm.WindowState != FormWindowState.Minimized)
+            {
+                Update();
+                Draw();
+                gfxScreenshot = Graphics.FromImage(bmpScreenshot);
+                IntPtr dc = gfxScreenshot.GetHdc();
+                PrintWindow(_dxForm.Handle, dc, 0);
+                gfxScreenshot.ReleaseHdc();
+                gfxScreenshot.Dispose();
+                _overlayTarget.Invalidate();
+            }
+            // Calculate Frame Limiting
+            //long end = DateTime.Now.Ticks;
+            //long duration = end - time;
+            int framerate = Settings.frameRate;
+            if (Settings.frameRate > 0)
+                Thread.Sleep(new TimeSpan(10000000 / Settings.frameRate));
+        }
+
+        class OverlayForm : Form
+        {
+            private int PreviousStyle;
+
+            public OverlayForm() : base()
+            {
+                DoubleBuffered = true;
+            }
+            public void SetFormTransparent()
+            {
+                IntPtr Handle = this.Handle;
+                PreviousStyle = GetWindowLong(Handle, GWL_EXSTYLE);
+                SetWindowLong(Handle, GWL_EXSTYLE, Convert.ToInt32(PreviousStyle | WS_EX_LAYERED | WS_EX_TRANSPARENT));
+            }
+
+            public void SetFormNormal()
+            {
+                IntPtr Handle = this.Handle;
+                SetWindowLong(Handle, GWL_EXSTYLE, Convert.ToInt32(PreviousStyle | WS_EX_LAYERED));
+            }
+        }

```

Comment:
NIT: The local variables `IntPtr Handle` are unnecessary. The second function can be shortened to a simple expression.

```C#
public void SetFormTransparent()
{
	PreviousStyle = GetWindowLong(Handle, GWL_EXSTYLE);
	SetWindowLong(Handle, GWL_EXSTYLE, Convert.ToInt32(PreviousStyle | WS_EX_LAYERED | WS_EX_TRANSPARENT));
}
public void SetFormNormal() =>
	SetWindowLong(Handle, GWL_EXSTYLE, Convert.ToInt32(PreviousStyle | WS_EX_LAYERED));
```